File: android/app/src/main/java/com/supersecurechat/app/SscDeepLink.kt
package com.supersecurechat.app

import android.content.Intent
import android.net.Uri

object SscDeepLink {
    private const val SCHEME = "ssc"
    private const val BUNDLED_BASE = "file:///android_asset/www"

    /**
     * Resolve an incoming [Intent] to a web URL loaded inside the bundled WebView shell.
     * Returns null when the intent carries no deep-link data that needs special routing.
     */
    fun resolveToWebUrl(intent: Intent?): String? {
        val data: Uri = intent?.data ?: return null
        if (data.scheme == SCHEME) {
            return when (data.host) {
                "link-device" -> {
                    val query = data.query?.let { "?$it" } ?: ""
                    "$BUNDLED_BASE/index.html#/link-device$query"
                }
                "add" -> {
                    val segment = data.pathSegments.firstOrNull() ?: ""
                    "$BUNDLED_BASE/index.html#/add/$segment"
                }
                "auth" -> "$BUNDLED_BASE/index.html#/auth${data.query?.let { "?$it" } ?: ""}"
                else -> null
            }
        }
        if (data.scheme == "https" && data.host == "www.supersecurechat.com") {
            return "$BUNDLED_BASE/index.html#${data.path ?: ""}"
        }
        return null
    }
}

File: ios/SuperSecureChat/SscDeepLink.swift
import Foundation

enum SscDeepLink {
    static func webPath(for url: URL) -> String? {
        if url.scheme == "ssc" {
            let host = url.host ?? ""
            let path = url.path.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
            if host == "link-device" { return "/link-device" }
            if host == "add", !path.isEmpty { return "/add/\(path)" }
            return "/"
        }
        if url.host == "www.supersecurechat.com", url.path.hasPrefix("/add/") {
            return url.path
        }
        return nil
    }
}